sandbox/apparmor/apparmor.go

```diff
@@ -396,6 +398,30 @@ func probeParserFeatures() ([]string, error) {
 	return features, nil
 }
 
+func systemAppArmorLoadsSnapPolicy() bool {
+	// on older Ubuntu systems the system installed apparmor may try and
+	// load snapd generated apparmor policy (LP: #2024637)
+	f, err := os.Open(filepath.Join(dirs.GlobalRootDir, "/lib/apparmor/functions"))
+	if err != nil {
+		logger.Debugf("cannot open apparmor functions file: %v", err)
```

Comment:
we probably shouldn't log if the error is the file not being there at all

Reply:
I added the logging, I wonder if I should just remove it?

Reply:
well, the question is a bit larger than logging, the question is also whether we should return true or false in the face of error. in principle we should return true not knowing, but then if the file is missing we might return false at least on != ubuntu

Reply:
this file is the one which contains the code to load snapd-generated apparmor profiles - and as far as I know it is only present on debian/ubuntu -  if we fail to open it then it either doesn't exist or something is really odd with permissions in which case it is likely not going to load any snapd apparmor profiles, so in this case I think returning false in all cases of error makes sense.